file/src/main/java/org/apache/calcite/adapter/file/CsvEnumerator.java

```diff
@@ -68,18 +69,16 @@
   private final RowConverter<E> rowConverter;
   private @Nullable E current;
 
-  private static final FastDateFormat TIME_FORMAT_DATE;
-  private static final FastDateFormat TIME_FORMAT_TIME;
-  private static final FastDateFormat TIME_FORMAT_TIMESTAMP;
+  private static final DateTimeFormatter TIME_FORMAT_DATE;
+  private static final DateTimeFormatter TIME_FORMAT_TIME;
+  private static final DateTimeFormatter TIME_FORMAT_TIMESTAMP;
```

Comment:
So `DateTimeFormatter` was added in Java 8. Can we use this? I.e. what's the minimum Java baseline for Calcite?

Reply:
We claimed support 8+ jdk in https://github.com/apache/calcite/blob/master/site/_docs/howto.md . So I think we can use it.

Reply:
It is a good idea to use `DateTimeFormatter` and we should do it. We should be a bit careful though cause every time there is a change in Date/Time/Timestamp parsing there are problems (many times backward compatibility) that go unnoticed.

Would it be possible to add some tests for dates/timestamps in the range `[0000-01-01 ,9999-12-29]` (a snippet generating random timestamps can be found [here](https://github.com/apache/hive/blob/3c9ea5d8d6ecc0d3749e22b28bfb7690dd3f1be1/ql/src/test/org/apache/hadoop/hive/ql/io/parquet/serde/TestParquetTimestampsHive2Compatibility.java#L118)) and check/document the differences with the old formatter. It's a bit unfortunate that we don't have such tests already.